msctl

```diff
@@ -1115,11 +1137,20 @@ stop() {
 # @param 1 The world server to forcibly stop.
 # ---------------------------------------------------------------------------
 forceStop() {
+  local WAIT
   # Try to stop the server cleanly first.
-  stop "$1"
-  sleep 5
+  sendCommand $1 "stop"
+  sendCommand $1 "end"
+  # Wait for the server to shut down.
+  WAIT=0
+  while serverRunning $1 && [ $WAIT -le 20 ]; do
```

Comment:
I like the additional wait time here to see if the server will shut off on it's own.